modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenTestCase.java

```diff
@@ -0,0 +1,577 @@
+        }
+
+        // Creates business API
+        BusinessAPICreationModel businessAPICreationModel = new BusinessAPICreationModel();
+        businessAPICreationModel.setName(externalServiceName);
+        businessAPICreationModel.setIdentifier(externalServiceURI);
+        businessAPICreationModel.setDescription("This is a test external service");
+        businessAPICreationModel.setRequiresAuthorization(REQUIRES_AUTHORIZATION);
+        List<ScopeGetModel> newScopes = new ArrayList<>();
+        customScopes.forEach(scope -> {
+            ScopeGetModel newCustomScope = new ScopeGetModel();
+            newCustomScope.setName(scope);
+            newCustomScope.setDescription("This is a test scope");
+            newCustomScope.setDisplayName(scope);
+            newScopes.add(newCustomScope);
+        });
+        businessAPICreationModel.setScopes(newScopes);
+        String businessAPIID = oAuth2Service.createBusinessAPIs(businessAPICreationModel);
+
+        // Authorizes business APIs
+        AuthorizedAPICreationModel authorizedBusinessAPICreationModel = new AuthorizedAPICreationModel();
+        authorizedBusinessAPICreationModel.setId(businessAPIID);
+        authorizedBusinessAPICreationModel.setPolicyIdentifier(RBAC);
+        authorizedBusinessAPICreationModel.setScopes(customScopes);
+        oAuth2Service.authorizeBusinessAPIs(applicationId, authorizedBusinessAPICreationModel);
+    }
+
+    /**
+     * Provides data for creating pre issue access token action
+     * Each dataset consists of an endpoint uri and authentication type.
+     *
+     * @return Two-dimensional array containing pairs of endpoint uris and authentication types.
+     */
+    @DataProvider(name = "getEndpointDetails")
+    public Object[][] getEndpointDetails() {
+
+        return new Object[][]{
+                {"https://wso2is.free.beeceptor.com", "Basic"}
+        };
+    }
+
+    @Test(groups = "wso2.is", description = "Create an action of type PreIssueAccessToken",
+            dataProvider = "getEndpointDetails", dependsOnMethods = "testRegisterPreRequisites")
+    private void testCreatePreIssueAccessTokenAction(String uri, String authenticationTypeString) {
+        Endpoint endpoint = new Endpoint();
+        endpoint.setUri(uri);
+        AuthenticationType authenticationType = new AuthenticationType();
+        switch (authenticationTypeString.toUpperCase()) {
+            case "BASIC":
+                authenticationType.setType(AuthenticationType.TypeEnum.BASIC);
+                Map<String, Object> authProperties = new HashMap<>();
+                authProperties.put(USERNAME_PROPERTY, TEST_USER);
+                authProperties.put(PASSWORD_PROPERTY, ADMIN_WSO2);
+                authenticationType.setProperties(authProperties);
+                break;
+            default:
+                authenticationType.setType(AuthenticationType.TypeEnum.NONE);
+        }
+        endpoint.setAuthentication(authenticationType);
+
```

Comment:
```suggestion
    private void testCreatePreIssueAccessTokenAction(String uri, AuthenticationType.TypeEnum authenticationType, Map<String, Object>authProperties) {
        Endpoint endpoint = new Endpoint()
                         .uri(uri)
                         .authentication(new AuthenticationType()
                                      .type(authenticationType)
                                      .properties(authProperties));

```

Reply:
Altered the implementation with 6e4aab9c2f07c6fa7ac9e7d13b2ea705224a25a5